pkg/frontend/v2/frontendv2pb/frontend.proto

```diff
@@ -30,4 +31,22 @@ message QueryResultRequest {
     // calling QueryResult, and that is where Frontend expects to find it.
 }
 
+message QueryResultStreamRequest {
+  uint64 queryID = 1;
+  oneof data {
+    QueryResultMetadata metadata = 2;
+    QueryResultBody body = 3;
+  }
+}
+
+message QueryResultMetadata {
+    int32 code = 1;
+    repeated httpgrpc.Header headers = 2;
+    stats.Stats stats = 3;
+}
+
+message QueryResultBody {
+    bytes chunk = 1;
```

Comment:
I'm wondering how @aknuds1 is planning to implement label values streaming. It would be nice to align the two approaches in terms of the data type before we merge the first one

Reply:
I'm actually only tackling ingester to querier label values streaming. The protobuf change can be seen [here](https://github.com/grafana/mimir/pull/5568/files#diff-5b6c87318632a5b2830782c3245ecaffa95a2c45d49cd155050078a05fca2c05R25).

Reply:
I think that the querier->frontend streaming from this PR could be reused for streaming label values responses into frontends as well by just adding the relevant internal header. I suppose more changes would be needed in the frontend itself for end-to-end streaming.